lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -2,6 +2,72 @@
 
 module PolicyMachineStorageAdapter
   class ActiveRecord
+    class PolicyElement
+
+      # Given a list of operation set ids and a list of operations
+      # Returns a map of operations to the operation set ids that contain them
+      def self.operation_sets_containing(operation_set_ids, operations)
+        query = <<~SQL
+          WITH RECURSIVE accessible_operations AS (
+            (
+              SELECT
+                child_id,
+                parent_id,
+                parent_id AS operation_set_id
+              FROM assignments
+              WHERE parent_id IN (?)
+            )
+            UNION ALL
+            (
+              SELECT
+                assignments.child_id,
+                assignments.parent_id,
+                accessible_operations.operation_set_id AS operation_set_id
+              FROM assignments
+              INNER JOIN accessible_operations
+                ON accessible_operations.child_id = assignments.parent_id
+            )
+          )
+
+          SELECT DISTINCT accessible_operations.operation_set_id, ops.unique_identifier
+          FROM accessible_operations
+          JOIN policy_elements ops
+            ON ops.id = accessible_operations.child_id
+          WHERE ops.unique_identifier IN (?)
```

Comment:
Does this do additional filtering not done in L18?
also, hello.

Reply:
oooo nvm youre traversing the operation_set graph

Reply:
lol howdy, i didn't realize this was a public gem 😉 .. yes, this is just a slight variation of `with_accessible_operation` down below

Reply:
Fiona!!!!!